packages/core/src/Repo.ts

```diff
@@ -419,20 +404,19 @@ export class Repo<T = any> {
     return this.storage.getDocumentChanges(documentId)
   }
 
-  /**
-   * Saves the snapshot for the given `documentId`, replacing any existing snapshot.
-   * @param documentId
-   * @param snapshot
-   */
+  /** Saves the snapshot for the given `documentId`, replacing any existing snapshot. */
   private async saveSnapshot(documentId: string, document: A.Doc<T>) {
     const snapshot: any = clone(document)
+    const clock = getClock(document)
+    this.updateClock(documentId, clock)
+
     if (snapshot[DELETED]) {
       this.removeSnapshot(documentId)
       await this.storage.deleteSnapshot(documentId)
     } else {
       this.log('saveSnapshot', documentId, document)
       this.setSnapshot(documentId, snapshot)
-      await this.storage.putSnapshot(documentId, snapshot)
+      await this.storage.putSnapshot(documentId, snapshot, clock)
```

Comment:
Any time the snapshot changes, it's safe to assume that the clock has changed as well; so we update both at the same time always.